tests/integration_tests/test_metadata_mapper.py

```diff
@@ -0,0 +1,308 @@
+import re
+
+import pytest
+
+from mandible.metadata_mapper import (
+    ConfigSourceProvider,
+    Context,
+    MetadataMapper,
+    MetadataMapperError,
+    PySourceProvider,
+    Source,
+)
+from mandible.metadata_mapper.format import Json, Xml
+from mandible.metadata_mapper.storage import LocalFile
+
+
+@pytest.fixture
+def mapper(config):
+    return MetadataMapper(
+        template=config["template"],
+        source_provider=ConfigSourceProvider(config["sources"])
+    )
+
+
+def test_empty_mapping_empty_context():
+    mapper = MetadataMapper({})
+    assert mapper.get_metadata(Context()) == {}
+
+
+def test_constant_mapping_empty_context():
+    template = {
+        "foo": "bar",
+        "baz": "qux"
+    }
+    mapper = MetadataMapper(template)
+
+    assert mapper.get_metadata(Context()) == template
+
+
+def test_empty_context(fixed_name_file_config):
+    mapper = MetadataMapper(
+        template={
+            "foo": {
+                "@mapped": {
+                    "source": "fixed_name_file",
+                    "key": "foo"
```

Comment:
It looks like there are a lot of missing trailing commas on this file. Some have it, some don't.